backend/s3/file.go

```diff
@@ -209,11 +209,16 @@ func (f *File) Close() error {
 		if err != nil {
 			return err
 		}
+
+		f.writeBuffer = nil
+		err = waitUntilFileExists(f, 5)
+		if err != nil {
+			return err
+		}
 	}
 
 	f.writeBuffer = nil
-
-	return waitUntilFileExists(f, 5)
+	return nil
 }
```

Comment:
setting writeBuffer to nil only needs to happen inside the `if f.writeBuffer != nil {` block (because otherwise it would already be nil).  I think you may have had it like this before but I think we can simplify this to:
```go
		}

		f.writeBuffer = nil
		return waitUntilFileExists(f, 5)
	}

	return nil
}
```